app/controllers/posts_controller.rb

```diff
@@ -10,14 +10,25 @@ def show
 
   def new
     @post = Post.new
+    @user = User.find(params[:user_id])
   end
 
   def create
-    @post = Post.new(post_params)
-    @post.author = current_user
+    @user = User.find(params[:user_id])
+    @post = @user.posts.new(
+      title: post_params[:title],
+      text: post_params[:text],
+      author_id: post_params[:user_id],
+      comments_counter: 0,
+      likes_counter: 0
+    )
+    # @post = Post.new(post_params)
+    # @post.author = current_user
 
     if @post.save
-      redirect_to current_user, notice: 'You have successfully created a post'
+      @post.update_posts_counter
+      redirect_to current_user
+      flash[:notice] = 'You have successfully created a post'
```

Comment:
- [ ] Flash message is not working. I can't see the message anywhere after creating the post. Kindly fix it.

Reply:
Addressed